File: src/main/java/hilfling/backend/hilfing/model/Album.java
package hilfling.backend.hilfing.model;

import lombok.Data;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Data
@Entity
@Table(name="album")
public class Album implements Serializable, BaseModel {
    public Album() {
    }

    public Album(String title, Boolean analog) {
        this.title = title;
        this.analog = analog;
        this.dateCreated = new Date();
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "title", nullable = false, unique = true)
    private String title;

    // TODO: Autogenerate
    @Column(name = "date_created", updatable = false)
    @ColumnDefault("CURRENT_TIMESTAMP")
    private Date dateCreated;

    @Column(name = "analog", nullable = false)
    @ColumnDefault("false")
    private Boolean analog;

}

File: src/main/java/hilfling/backend/hilfing/model/Album.kt
package hilfling.backend.hilfing.model

import me.liuwj.ktorm.database.Database
import me.liuwj.ktorm.entity.Entity
import me.liuwj.ktorm.entity.sequenceOf
import me.liuwj.ktorm.schema.*
import java.time.LocalDate

interface Album : Entity<Album> {
    companion object : Entity.Factory<Album>()

    val id : Long
    var title: String
    var dateCreated : LocalDate
    var isAnalog : Boolean
}

object Albums : Table<Album>("t_album") {
    val id = long("id").primaryKey().bindTo { it.id }
    val title = varchar("title").bindTo { it.title }
    val dateCreated = date("date_created").bindTo { it.dateCreated }
    val isAnalog = boolean("is_analog").bindTo { it.isAnalog }
}

val Database.albums get() = this.sequenceOf(Albums)